src/Game.py

```diff
@@ -0,0 +1,332 @@
+
+    #   a function that returns whether the game should continue to be paused
+    #
+    #   return : a boolean that answers 'Is the game still paused ?'
+    def isWaiting(self):
+        pacman = self.entitySet.pacman
+        # pacman : an instance of the PacMan class
+        
+        waiting = True
+        dir = pacman.dir
+        for event in pg.event.get():
+            self.exitCheck(event)
+            if waiting :
+                if event.type == pg.KEYDOWN :
+                    if event.key in Player.KEYS:
+                        pacman.dir = Player.KEYS.index(event.key)
+                    waiting = self.isDirWall()
+        if waiting : 
+            pacman.dir = dir
+        return waiting
+
+    #   a function that returns if the player is still playing
+    #
+    #   return : a boolean that answers 'Is the player still playing ?'
+    def isPlaying(self):
+        player = self.player
+        pacman = self.entitySet.pacman
+        dir = pacman.dir
+        # player : an instance of the Player class
+        # pacman : an instance of the PacMan class
+        # dir : direction of Pacman
+
+        newDir = dir
+        playing = True
+        for event in pg.event.get():
+            self.exitCheck(event)
+
+            # a key has been pressed
+            if event.type == pg.KEYDOWN:
+                
+                # a pause key
+                if event.key in Player.PAUSE_KEYS:
+                    playing = False
+
+                # a directional key
+                elif event.key in Player.KEYS:
+                    pacman.dir = Player.KEYS.index(event.key)
+                    player.keyBools[3-pacman.dir] = False
+
+                    # no wall in this direction
+                    if not self.isDirWall():
+                        newDir = pacman.dir
+                        player.keyBools[pacman.dir] = False
+
+                    # a wall in this direction
+                    else:
+                        player.keyBools[pacman.dir] = True
+
+            # a key has been released
+            elif event.type == pg.KEYUP and event.key in Player.KEYS:
+                player.keyBools[Player.KEYS.index(event.key)] = False
+
+        # no pause requested 
+        if playing:
+            if dir == newDir:
+                for altDir in [3-(1-dir)%4, (1-dir)%4, 3-dir]:
+                    pacman.dir = altDir
+                    if player.keyBools[altDir] and not self.isDirWall():
+                        newDir = altDir
+                        
+            pacman.dir = newDir
+            player.keyBools[pacman.dir] = False
+        
+        # a pause requested
+        else:
+            player.keyBools = [False, False, False, False]
+        
+        return playing
+
+    #TODO : process the case where the ghosts has bean eaten
+    #   a function to move the movables of the game
+    def move(self):
+        walls = self.map.walls
```

Comment:
Shouldn't it be set at init as the move function will be called at each frame ?

Reply:
I don't think it's that important, between `self.walls` and `self.map.walls` there is only a small gain plus the cost of `self.map.walls` is not that high

Reply:
Alright then it's all good for me !